pss/message/message_test.go

```diff
@@ -0,0 +1,51 @@
+package message_test
+
+import (
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/epiclabs-io/ut"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethersphere/swarm/pss/message"
+)
+
+func TestMessage(tx *testing.T) {
+	t := ut.BeginTest(tx, false) // set to true to generate test results
+	defer t.FinishTest()
+
+	// generate some test messages deterministically
+	for i, topicString := range someTopics {
+		flags := message.Flags{
+			Raw:       i&0x1 == 0,
+			Symmetric: i&0x3 == 0,
+		}
+
+		msg := message.New(flags)
+		msg.To = ut.RandomArray(i, common.AddressLength)
```

Comment:
Please consider sourcing address from already existing methods that generate addresses, for example `network.RandomAddress()` (which in turn could use a deterministic seed as you use here in `RandomArray`)

Or even better would be if we had an address constructor in `pss` itself, since `pss` also operates with partial addressing.

I would prefer if the different addresses we use in Swarm are _strongly_ typed.

Reply:
`pot.RandomAddress()` does not currently generate addresses based on a seed, and I need that for deterministic testing, hence the use of `RandomArray()` for now.

Opened #1735 to track a change to that method so it can be used.

> Or even better would be if we had an address constructor in pss itself, since pss also operates with partial addressing.

We can visit this when visiting `PssAddress` during the refactor.

> I would prefer if the different addresses we use in Swarm are strongly typed.

Me too, but I try to keep the scope of the refactor to `PssMsg` exclusively, otherwise the PR scope creeps and then it becomes a big PR.